stores/slabbuffer.go

```diff
@@ -378,11 +382,15 @@ func (buf *SlabBuffer) acquireForUpload(lockingDuration time.Duration) bool {
 	return true
 }
 
-func (buf *SlabBuffer) recordAppend(data []byte, mustFit bool, minShards uint8) (object.SlabSlice, []byte, bool, error) {
+func isCompleteBuffer(size, maxSize, completionThreshold int64) bool {
```

Comment:
I would make this a method on the `SlabBuffer` and also add another one called `canFit` so the `if-else` below becomes `if buf.isComplete() { ...} else if buf.canFit() { ... } else if !mustFit {}`